mongo/integration/log_helpers_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (C) MongoDB, Inc. 2023-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package integration
+
+import (
+	"context"
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/internal/logger"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+)
+
+type testLogSink struct {
+	logs       chan func() (int, string, []interface{})
+	bufferSize int
+	logsCount  int
+	errsCh     chan error
+}
+
+type logValidator func(order int, lvl int, msg string, kv ...interface{}) error
+
+func newTestLogSink(ctx context.Context, mt *mtest.T, bufferSize int, validator logValidator) *testLogSink {
+	mt.Helper()
+
+	sink := &testLogSink{
+		logs:       make(chan func() (int, string, []interface{}), bufferSize),
+		errsCh:     make(chan error, bufferSize),
```

Comment:
```suggestion
		errsCh:     make(chan error, 1),
```
Can we just make a buffered channel of size 1? We don't seem to care about errors beyond the first one sent to the channel.

Reply:
Even though we only care bout one error we need to buffer the errors to the size of the clients, to avoid blocking errors in the worker.